snapcraft/plugins/_python/_pip.py

```diff
@@ -129,10 +137,23 @@ def setup(self):
         Check to see if pip has already been installed. If not, fetch pip,
         setuptools, and wheel, and install them so they can be used.
         """
-        # Check to see if we have our own pip, yet. If not, we need to use the
-        # pip on the host (installed via build-packages) to grab our own.
-        if not self.is_setup():
-            logger.info('Fetching pip, setuptools, and wheel...')
+
+        self._ensure_pip_installed()
+        self._ensure_wheel_installed()
+        self._ensure_setuptools_installed()
+
+    def is_setup(self):
+        """Return true if this class has already been setup."""
+
+        return (
+            self._is_pip_installed() and self._is_wheel_installed() and
+            self._is_setuptools_installed())
+
+    def _ensure_pip_installed(self):
+        # Check to see if we have our own pip. If not, we need to use the pip
+        # on the host (installed via build-packages) to grab our own.
+        if not self._is_pip_installed():
+            logger.info('Fetching pip...')
```

Comment:
A nit, but maybe this log entry would be more accurate as `Fetching and installing pip`. Or have two log entries, one for download and one for install.

Reply:
Yeah good call, I went with "fetching and installing" instead of multiple prints. Done.